motionplan/check.go

```diff
@@ -198,20 +206,36 @@ func checkPlanRelative(
 		return err
 	}
 
-	// create a list of segments to iterate through
 	segments := make([]*ik.Segment, 0, len(plan.Path())-wayPointIdx)
 	// get checkFrame's currentInputs
 	// *currently* it is guaranteed that a relative frame will constitute 100% of a solver frame's dof
 	checkFrameCurrentInputs, err := sf.mapToSlice(currentInputs)
 	if err != nil {
 		return err
 	}
-	arcEndInputs, err := sf.mapToSlice(plan.Trajectory()[wayPointIdx])
+
+	poses, err := plan.Path().GetFramePoses(checkFrame.Name())
+	if err != nil {
+		return err
+	}
+	currentWayPointTraj := plan.Trajectory()[wayPointIdx]
+	currentWayPointTraj[localizationFrame.Name()] = referenceframe.FloatsToInputs([]float64{
```

Comment:
IMO it would make more sense for this to live in the motion service. I think if we do that, it removes the need to pass in the localization frame specially.

Reply:
hmmmmm, this is an interesting suggestion. 


The first route I took to attempt at accomplishing this suggestion was to augment the existing plan object we pass into `CheckPlan` to house inputs information relating to the `LocalizationFrame`.

However, I do not believe that it is possible to accomplish this (this way) due to:
```
		nextInputs[localizationFrame.Name()] = startInputs[localizationFrame.Name()]
```
Above, is a part of the for loop we construct to create the necessary segments to iterate over.
Since we want our `nextInputs` to equal `startInputs` for the LocalizationFrame, there is no way AFAIK to augment the existing plan to accomplish this.